x/genutil/collect.go

```diff
@@ -134,9 +134,6 @@ func CollectTxs(cdc codec.JSONMarshaler, txJSONDecoder sdk.TxDecoder, moniker, g
 
 		// genesis transactions must be single-message
 		msgs := genTx.GetMsgs()
-		if len(msgs) != 1 {
-			return appGenTxs, persistentPeers, errors.New("each genesis transaction must provide a single genesis message")
-		}
 
 		// TODO abstract out staking message validation back to staking
 		msg := msgs[0].(*stakingtypes.MsgCreateValidator)
```

Comment:
I see we're casting it to CreateValidator, shouldn't we change this if we want to support different messages?

Reply:
I think we should at least check the type assertion if it is ok.

Reply:
Out of scope of this PR, lets open an issue with this